apps/web/src/state/farmsV4/state/poolApr/atom.ts

```diff
@@ -10,31 +10,58 @@ export const poolsAtom = atom<PoolInfo[]>((get) => {
   return farmPools.concat(extendPools)
 })
 
-export type PoolAprDetail = {
-  lpApr: {
-    value: `${number}`
-  }
-  cakeApr: {
-    // default apr
-    value: `${number}`
-    // apr with boost, not related to user account
-    boost?: `${number}`
-  }
-  // @todo @ChefJerry implement merklApr
-  merklApr: {
-    value: `${number}`
-  }
+export type AprValue = {
+  [key: ChainIdAddressKey]: `${number}`
 }
+export type MerklApr = AprValue
+export const merklAprAtom = atom<MerklApr>({})
+
+export type LpApr = AprValue
+export const lpAprAtom = atom<LpApr>((get) => {
+  const pools = get(poolsAtom)
+  return pools.reduce((acc, pool) => {
+    return {
+      ...acc,
+      [`${pool.chainId}:${pool.lpAddress}`]: pool.lpApr ?? '0',
+    }
+  }, {} as LpApr)
+})
+
+export type CakeApr = Record<ChainIdAddressKey, { value: `${number}`; boost?: `${number}` }>
+export const cakeAprAtom = atom<CakeApr>({})
+
+export type PoolApr = Record<
+  ChainIdAddressKey,
+  {
+    lpApr: `${number}`
+    cakeApr: {
+      value: `${number}`
+      boost?: `${number}`
+    }
+    merklApr: `${number}`
+  }
+>
 
-export type PoolApr = Record<ChainIdAddressKey, PoolAprDetail>
+export const poolAprAtom = atom<PoolApr>((get) => {
+  const lpAprs = get(lpAprAtom)
+  const cakeAprs = get(cakeAprAtom)
+  const merklAprs = get(merklAprAtom)
```

Comment:
I think this way may not friendly to extends new apr,
how about merge all xxApr into a central config
think about there is a new xApr need to added, if we can just implement a hook to getXApr, and add to config like something about: { xApr: getXApr }, and then all is done, it'll be better